### PR TITLE
cam6_3_002: Chemistry code changes for 'weak scaling' issues

### LINKED ISSUE #219: Chemistry code changes for scaling to high-res grids
These phys_grid interfaces will go away:
 scatter_field_to_chunk
 get_lat_all_p 
 get_lon_all_p 
 get_lat_p
 get_lon_p

This requires refactoring some code and removing code that is no longer needed.

### PR DESCRIPTION
**Pull request by fvitt**
This removes this use of phys_grid interfaces in chemistry code that will become obsolete in the near future.  The phys_grid interfaces that will go away are:
 scatter_field_to_chunk
 get_lat_all_p
 get_lon_all_p
 get_lat_p
 get_lon_p

These phys_grid routines will be removed in an effort to allow CAM to scale better to high-resolution grids

This PR also removes chemistry code that is no longer used.

closes #219

### CODE REVIEW

File: bld/namelist_files/namelist_definition.xml [hunk 1]
```diff
@@ -5374,31 +5374,9 @@ Full pathname of boundary dataset for airplane emissions.
...
-Full pathname of dataset which contains land vegitation information used in 'xactive_atm'
-method of calculating dry deposition of chemical tracers.
-Default: set by build-namelist.
-</entry>
-
-<entry id="season_wes_file" type="char*256" input_pathname="abs" category="cam_chem"
-       group="chem_inparm" valid_values="" >
-Full pathname of dataset which contains season information used in 'xactive_atm'
-method of calculating dry deposition of chemical tracers.
+Full pathname of dataset which contains land vegitation information used in modal aerosol deposition.
```
Comment (nusbaume) [inline bld/namelist_files/namelist_definition.xml:5379]: Typo here (`vegetation`).

File: src/chemistry/mozart/mo_drydep.F90 [hunk 2]
```diff
@@ -2328,34 +795,44 @@ subroutine interp_map( plon, plat, nlon_veg, nlat_veg, npft_veg, lat_veg, lat_ve
...
-    ! 	... make sure there are no out of range values
-    !-------------------------------------------------------------------------------------
-    where (fraction_landuse < 0._r8) fraction_landuse = 0._r8
-    where (fraction_landuse > 1._r8) fraction_landuse = 1._r8
+    
+    do lchnk = begchunk, endchunk
+       ncol = get_ncols_p(lchnk)
+       do icol= 1,ncol
+          call get_rlat_all_p(lchnk, ncol, rlats(:ncol))
+          call get_rlon_all_p(lchnk, ncol, rlons(:ncol))
```
Comment (nusbaume) [inline src/chemistry/mozart/mo_drydep.F90:805]: Should these two calls be outside the `icol` loop?

**Review by nusbaume — CHANGES_REQUESTED** [review]
Generally looks fine to me, but I do have some (hopefully minor) requests and questions.

**Comment by gold2718** [comment]
@fvitt, can you merge this up to cam6_3_000 to remove conflicts and make it easier to compare?

**Review by brian-eaton — APPROVED** [review]

**Review by gold2718 — APPROVED** [review]
Wow, lots of cleanup.
Looks okay o me.

**Review by nusbaume — APPROVED** [review]
Thanks for taking care of my requests!  Everything looks good to me now.

File: src/chemistry/mozart/mo_drydep.F90 [hunk 3]
```diff
@@ -299,1301 +220,93 @@ subroutine drydep_fromlnd( ocnfrac, icefrac, ncdate, sfc_temp, pressure_sfc,  &
...
-    endif
-    if( nc4ch2oh_dd ) then
-       if( map(nc4ch2oh_ndx) == 0 ) then
-          depvel(:ncol,nc4ch2oh_ndx) = depvel(:ncol,h2o2_ndx)
-          dflx(:ncol,nc4ch2oh_ndx)   = wrk(:ncol) * depvel(:ncol,nc4ch2oh_ndx) * q(:ncol,plev,nc4ch2oh_ndx)
-       end if
+    
+    !-------------------------------------------------------------------------------------
+    !        ... assign CO tags to CO
+    ! put this kludge in for now ...
```
Comment (cacraigucar) [inline src/chemistry/mozart/mo_drydep.F90:255]: Is this still a kludge?  What will it take to make it correct?  Does this need an issue opened for it or should the comment be modified?

**Review by cacraigucar — APPROVED** [review]

File: src/chemistry/mozart/mo_drydep.F90 (hunk 3 above)
Comment (fvitt) [reply src/chemistry/mozart/mo_drydep.F90:255]: > Is this still a kludge? What will it take to make it correct? Does this need an issue opened for it or should the comment be modified?

Yes this is still a kludge.  There will be a larger effort to refactor how dry deposition implemented in the not too distant future.  Hopefully we will remove these kludges then.  I propose we leave the comment as it is for this merge.

**Pull request merged**